src/transformers/tokenization_prophetnet.py

```diff
@@ -0,0 +1,170 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import List, Optional
+
+from .tokenization_bert import BertTokenizer
+from .tokenization_utils import PreTrainedTokenizer
+from .tokenization_xlm_roberta import XLMRobertaTokenizer
+
+
+logger = logging.getLogger(__name__)
+
+VOCAB_FILES_NAMES = {"vocab_file": "prophetnet.tokenizer"}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        "microsoft/prophetnet-large-uncased": "https://s3.amazonaws.com/models.huggingface.co/bert/microsoft/prophetnet-large-uncased/prophetnet.tokenizer",
+        "microsoft/xprophetnet-large-wiki100-cased": "https://cdn.huggingface.co/microsoft/xprophetnet-large-wiki100-cased/prophetnet.tokenizer",
+    }
+}
+
+PRETRAINED_INIT_CONFIGURATION = {
+    "microsoft/prophetnet-large-uncased": {"do_lower_case": True, "xprophetnet_tokenizer": False},
+    "microsoft/xprophetnet-large-wiki100-cased": {"do_lower_case": False, "xprophetnet_tokenizer": True},
+}
+
+PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {
+    "microsoft/prophetnet-large-uncased": 512,
+    "microsoft/xprophetnet-large-wiki100-cased": 512,
+}
+
+
+class ProphetNetTokenizer(PreTrainedTokenizer):
+    r"""
+    ProphetNet inherit from BERT-tokenizer, xProphetNet  inherit from XLMR-tokenizer
+    """
+
+    vocab_files_names = VOCAB_FILES_NAMES  # default english version rather than cross-lingual version.
+    pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
+    pretrained_init_configuration = PRETRAINED_INIT_CONFIGURATION
+    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
+
+    def __new__(cls, **kwargs):
+        xprophetnet_tokenizer = (
+            False if "xprophetnet_tokenizer" not in kwargs.keys() else kwargs["xprophetnet_tokenizer"]
+        )
+        if xprophetnet_tokenizer:
+            super_class = XLMRobertaTokenizer
+        else:
+            super_class = BertTokenizer
+        cls = type(cls.__name__, (cls, super_class), {})
+        return super(ProphetNetTokenizer, cls).__new__(cls)
+
+    def __init__(
+        self,
+        vocab_file,
+        xprophetnet_tokenizer=False,
+        do_lower_case=None,
+        do_basic_tokenize=None,
+        never_split=None,
+        unk_token=None,
+        sep_token=None,
+        pad_token=None,
+        cls_token=None,
+        mask_token=None,
+        tokenize_chinese_chars=None,
+        bos_token=None,
+        eos_token=None,
+        **kwargs
+    ):
+        if not xprophetnet_tokenizer:
+            # inherit from BERT tokenizer
+            do_lower_case = True if do_lower_case is None else do_lower_case
+            do_basic_tokenize = True if do_lower_case is None else do_basic_tokenize
```

Comment:
@qiweizhen 

Is this line correct here or should it rather be:
```
do_basic_tokenize = True if do_basic_tokenize is None else do_basic_tokenize
```
?

Changing the line to the suggestion above would change the integration test. 
I guess it's a typo though correct?

Reply:
My mistike. You are correct. Thanks!

Reply:
No worries! I corrected the integration test.